tests/junit.py

```diff
@@ -26,6 +26,7 @@ def create_argparser() -> argparse.ArgumentParser:
     parser.add_argument('-n', '--nproc', type=int, default=1, help='number of MPI processes')
     parser.add_argument('-o', '--output', type=Optional[Path], default=None, help='Output file to write test')
     parser.add_argument('-b', '--junit-batch', type=str, default='', help='Name of JUnit batch for output file')
+    parser.add_argument('-np', '--pool-size', type=int, default=1, help='Number of test cases to run in parallel')
```

Comment:
I'm not sure what these change to `junit.py` and `junit_common.py` are, but they shouldn't be here. These lines of code are currently in `main`, so I'm not sure why they're showing up here. 

I think this just needs to be rebased onto `main`?